apis/kubevault/v1alpha1/types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	ofst "kmodules.xyz/offshoot-api/api/v1"
+)
+
+// +kubebuilder:validation:Enum=Provisioning;DataRestoring;Ready;Critical;NotReady;Halted;Sealed;Unsealed;Initializing;Initialized
+type VaultServerPhase string
+
+const (
+	// used for VaultServer that are currently provisioning
+	VaultServerPhaseProvisioning VaultServerPhase = "Provisioning"
+	// used for VaultServer for which data is currently restoring
+	VaultServerPhaseDataRestoring VaultServerPhase = "DataRestoring"
```

Comment:
Remove unused Phases, for now. Add them in a separate PR while implementing that feature.